contents/docs/cdp/common-questions.md

```diff
@@ -0,0 +1,49 @@
+---
+title: Troubleshooting and FAQs
+---
+
+## My app isn't working, what do I do?
+
+1. Checking that the app is enabled with the correct configuration options in the [data pipeline tab](https://us.posthog.com/apps). You can find the correct configuration options in the [app's documentation](/docs/cdp) by searching for it in the destinations, transformations, or filtering sections of the docs.
+2. Click "Logs & metrics" and go to the `processEvent metrics` tab to check that the app is processing events without errors.
+3. Check the `Logs` tab to see if there are any errors.
+4. Go to the data management tab to check if there any [ingestion warnings](https://us.posthog.com/data-management/ingestion-warnings).
+5. If the app relates to an external service, check that the external service is working correctly. Make a request to the relevant API endpoint or use a tool like [webhook.site](https://webhook.site/).
+
+## How do I capture data from another application?
+
+We deprecated the functionality of PostHog apps that enable you to capture data from other tools. Functionally, these apps were a cron which ran every minute (or more) and pulled data from another tool.
+
+There are two options for recreating this functionality:
+
+1. **Set up a cron of your own**. You can use a tool like [val.town](https://val.town/) to easily set up scheduled tasks that capture data from other tools and send it to PostHog. Our tutorials on [capturing new RSS items](/tutorials/rss-item-capture) and [events from Calendly webhooks](/tutorials/calendly-webhooks) are examples of this.
+
+2. **Use our data warehouse**. PostHog's data warehouse enables you to import and use data from apps like Stripe, Hubspot, and Postgres natively in PostHog. See our [docs](/docs/data-warehouse) for more information.
+
+## How do I track when users do an event for the first time?
+
+There are a few ways to capture a user doing an event for the first time. We cover a few of these in our tutorials on [tracking new and returning users](/tutorials/track-new-returning-users) and [first touch attribution](/tutorials/first-last-touch-attribution).
+
+- Create a [cohort](/docs/data/cohorts) matching users who have done an event for the first time recently.
+- Use a custom event to set a [person](/docs/getting-started/user-properties) or [event property](/docs/data/events). For example, you can set a `first_seen` property with the `$set_once` option.
+- Use [HogQL](/docs/product-analytics/sql) to query for the `min(timestamp)` of an event.
+
+## How do I migrate 20M+ events or rows into PostHog?
+
+If you're migrating a large amount of data (20 million rows or 10k requests per minute), please contact [sales@posthog.com](mailto:sales@posthog.com) to make sure you aren't rate limited.
+
+See [our migration docs](/docs/migrate/ingest-historic-data) for more information.
```

Comment:
cc @bretthoerner  - we tell them to add the historical flag already, correct? Let's document this here or in the linked doc

```suggestion
## How do I migrate events or rows into PostHog?

TODO: set the historical flag on your batches

If you're migrating a large amount of data (20 million rows or 10k requests per minute), please contact [sales@posthog.com](mailto:sales@posthog.com) to make sure you aren't rate limited.

See [our migration docs](/docs/migrate/ingest-historic-data) for more information.
```

Reply:
https://github.com/PostHog/posthog.com/pull/7917 is the flag everyone should use